packages/components/src/sharedHelpers/types.ts

```diff
@@ -9,6 +15,280 @@ export interface CommonAtlantisProps {
   id?: string;
 }
 
+/**
+ * Core ARIA attributes for input elements.
+ * Uses camelCase naming for consistency with React props pattern.
+ * Based on React's canonical ARIA type definitions.
+ */
+export interface AriaInputProps {
+  /**
+   * Defines a string value that labels the current element.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-label}
+   */
+  readonly ariaLabel?: AriaAttributes["aria-label"];
+
+  /**
+   * Identifies the element (or elements) that labels the current element.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-labelledby}
+   */
+  readonly ariaLabelledBy?: AriaAttributes["aria-labelledby"];
+
+  /**
+   * Identifies the element (or elements) that describes the object.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-describedby}
+   */
+  readonly ariaDescribedBy?: AriaAttributes["aria-describedby"];
+
+  /**
+   * Identifies the element (or elements) that provide a detailed, extended description.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-details}
+   */
+  readonly ariaDetails?: AriaAttributes["aria-details"];
+
+  /**
+   * ID of the currently active descendant element.
+   * Used for composite widgets like combobox or listbox.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-activedescendant}
+   */
+  readonly ariaActiveDescendant?: AriaAttributes["aria-activedescendant"];
+
+  /**
+   * Indicates the element that controls the current element.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-controls}
+   */
+  readonly ariaControls?: AriaAttributes["aria-controls"];
+
+  /**
+   * Indicates whether the element is expanded or collapsed.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-expanded}
+   */
+  readonly ariaExpanded?: AriaAttributes["aria-expanded"];
+
+  /**
+   * Indicates the type of autocomplete interaction.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-autocomplete}
+   */
+  readonly ariaAutocomplete?: AriaAttributes["aria-autocomplete"];
+
+  /**
+   * Indicates that user input is required before form submission.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-required}
+   */
+  readonly ariaRequired?: AriaAttributes["aria-required"];
```

Comment:
this one is a little weird

we don't have a required prop. that's not how you implement required on our input v2s, so there's no through line from that prop to this

however; marking an input with aria-required is a good idea